scene/resources/3d/primitive_meshes.cpp

```diff
@@ -194,6 +195,9 @@ Mesh::PrimitiveType PrimitiveMesh::surface_get_primitive_type(int p_idx) const {
 }
 
 void PrimitiveMesh::surface_set_material(int p_idx, const Ref<Material> &p_material) {
+	if (p_material == material) {
+		return;
+	}
```

Comment:
This check is redundant because `set_material` already calls it

Reply:
Thanks! I was waiting too make sure no one had any suggestions to squash the commits. I'll make the change and squash them now!